src/main/kotlin/org/rust/lang/core/completion/RsCompletionConfidence.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.completion
+
+import com.intellij.codeInsight.completion.CompletionConfidence
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.util.ThreeState
+import org.rust.lang.core.psi.RsElementTypes.IDENTIFIER
+import org.rust.lang.core.psi.RsLetDecl
+import org.rust.lang.core.psi.RsPatBinding
+import org.rust.lang.core.psi.ext.elementType
+import org.rust.lang.core.psi.ext.topLevelPattern
+
+class RsCompletionConfidence : CompletionConfidence() {
```

Comment:
Maybe it's better to check if the crate suppresses any of the standard non_camel_case_types/non_snake_case/etc warnings?

Reply:
I don't think this is helpful. Even if warnings are disabled, it is still unlikely that the user wants to do destructuring there. At first, I was going to not show this popup in all cases, and only then I came up with this heuristic. That is, if there is no heuristic, then in my opinion it is still better not to show the completion popup at all. A user can always call the completion manually in that 0.1% of cases when they needs it